### PR TITLE
Task 6

### PR DESCRIPTION
**Pull request by LeonidShatilo**
### Task:
Based on your previous tasks integrate Redux for state management (you need both redux and react-redux dependencies).
Store all the application data into Redux state (e.g. cards data, user data, authorization status).

### Implementation notes:

1. Create an additional folder in your app (store, redux, or something else). This folder will contain the store of your whole application (`createStore()`), all the required reducers for your app (if you have more than 1 reducer, don’t forget to **[combine](https://redux.js.org/api/combinereducers#combinereducersreducers)** them), contain all the required actions for your app (**type** is the only required field for **[action](https://redux.js.org/tutorials/fundamentals/part-2-concepts-data-flow#actions)**, other fields are up to you). You can create additional folders and files there if you need them.
2. For using Redux with React you will need the **[react-redux](https://react-redux.js.org/)** library. Don’t forget to wrap the entire components hierarchy `<Provider />` component and pass the store as a prop to it.
3. Each component, which needs access to a redux store can be wrapped with **[connect(mapStateToProps, mapDispatchToProps)(YourComponent)](https://react-redux.js.org/api/connect)** or you can use hooks (**[useDispatch](https://react-redux.js.org/api/hooks#usedispatch)**, **[useSelector](https://react-redux.js.org/api/hooks#useselector)**).
4. Use **[redux-thunk (purpose of redux-thunk, how to apply it to your store)](https://stackoverflow.com/questions/35411423/how-to-dispatch-a-redux-action-with-a-timeout/35415559#35415559)** as middleware in order to «teach» Redux to recognize special kinds of actions that are in fact functions. This middleware will help you to write the following «action-function»:

### CODE REVIEW

File: src/store/reducers/cardReducer.js [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      return payload;
+
+    case ADD_CARD:
+      return [...state, payload];
+
+    case REMOVE_CARD:
+      const id = payload;
+      const filteredCards = state.filter((card) => card.id !== id);
+
+      return [...filteredCards];
```
Comment (broshkabro) [inline src/store/reducers/cardReducer.js:17]: making a copy of filtered cards is redundant, the `filter `method creates a new array anyway

File: src/store/reducers/signReducer.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import { SIGN_IN, SIGN_OUT } from '../types/actionsTypes';
+
+const authToken = 'a8ffb78c-1e35-415b-a781-556541ad5cc8';
+const initialState = {
+  isSignIn: localStorage.getItem('authToken') === authToken ? true : false,
```
Comment (broshkabro) [inline src/store/reducers/signReducer.js:5]: redundant condition, the result of this expression `localStorage.getItem('authToken') === authToken` is a boolean

File: src/store/types/actionsTypes.js [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+export const SIGN_IN = 'SIGN/SIGN_IN';
+export const SIGN_OUT = 'SIGN/SIGN_OUT';
+
+export const SET_IS_SUBMIT = 'SUBMIT/SET_IS_SUBMIT';
+
+export const SET_EMPTY_CARD_INPUT = 'INPUT/SET_EMPTY_CARD_INPUT';
+export const SET_EMPTY_LOGIN_INPUT = 'INPUT/SET_EMPTY_LOGIN_INPUT';
+
+export const SHOW_SPINNER = 'SPINNER/SHOW_SPINNER';
+export const HIDE_SPINNER = 'SPINNER/HIDE_SPINNER';
```
Comment (broshkabro) [inline src/store/types/actionsTypes.js:14]: usually, we do not add a prefix (like 'SPINNER/', 'SUBMIT/') to the action name

**Review by broshkabro — COMMENTED** [review]
awesome

**Review by broshkabro — APPROVED** [review]

File: src/store/reducers/cardReducer.js (hunk 1 above)
Comment (LeonidShatilo) [reply src/store/reducers/cardReducer.js]: Right! I fixed it.

File: src/store/reducers/signReducer.js (hunk 2 above)
Comment (LeonidShatilo) [reply src/store/reducers/signReducer.js]: Oh, right! I removed the redundant condition. Thank you!

File: src/store/types/actionsTypes.js (hunk 3 above)
Comment (LeonidShatilo) [reply src/store/types/actionsTypes.js]: Thank you, I didn't know that.

**Pull request merged**